src/expr/src/scalar/func.rs

```diff
@@ -2850,6 +2850,35 @@ fn substr<'a>(datums: &[Datum<'a>]) -> Datum<'a> {
     }
 }
 
+fn split_part<'a>(datums: &[Datum<'a>]) -> Result<Datum<'a>, EvalError> {
+    let string = datums[0].unwrap_str();
+    let delimiter = datums[1].unwrap_str();
+
+    // If the provided delimiter is the empty string,
+    // PostgreSQL does not break the string into individual
+    // characters. Instead, it generates the following parts: [string].
+    let parts = if delimiter == "" {
+        vec![string]
+    } else {
+        string.split(delimiter).collect()
```

Comment:
This vec is going to allocate, which would be nice to avoid to make things a bit faster!

I think you could use `iter::nth` to avoid the allocation?

```
if delimiter == "" {
    return Ok(Datum::String(""));
}

let parts = string.split(delimiter);

// ...

Ok(Datum::String(parts.nth(usize_index).unwrap_or("")))
```

Reply:
Yes - thanks for sharing!